### PR TITLE
Sitreport We got incoming enemy ships

### PR DESCRIPTION
**Pull request by agrrr3**
* Defined via a tech spec via custom_sitreps.txt
* I need to test this a bit
  
[Forum discussion](http://freeorion.org/forum/viewtopic.php?f=6&t=10839)

### CODE REVIEW

File: default/customizations/custom_sitreps.txt [hunk 1]
```diff
@@ -105,6 +105,51 @@ EffectsGroup
...
+        System
+        // The following assets need protection
+        Contains And [
+            OwnedBy empire = Source.Owner
+            Or [
+                Planet
+                Ship
+            ]
+        ]
+        // Dont warn about system where there are already enemies this turn
```
Comment (geoffthemedio) [inline default/customizations/custom_sitreps.txt:119]: Do you want to never warn if there are any enemies in the system... or:
-Enemies with the same owner as those arriving
-Enemies visible to the owner the recipient of the message

File: default/customizations/custom_sitreps.txt [hunk 2]
```diff
@@ -105,6 +105,51 @@ EffectsGroup
...
+        Not Contains And [
+            Ship
+            Or [
+                OwnedBy affiliation = EnemyOf empire = Source.Owner
+                UnOwned
+            ]
+        ]
+        // Only warn if enemies are incoming
```
Comment (geoffthemedio) [inline default/customizations/custom_sitreps.txt]: Should probably check that the incoming enemies are visible to the recipient.

File: default/customizations/custom_sitreps.txt (hunk 1 above)
Comment (agrrr3) [reply default/customizations/custom_sitreps.txt:119]: The warning is to draw attention to the system. If there was already combat, you have already a message for doing that.

I didnt know that i have to script a visibility check. If I remember correctly, without OwnedBy empire = Source.Owner i did not get matches outside my known empire without checking visibility, is that the way it works? Does not sound right.

Comment (geoffthemedio) [reply default/customizations/custom_sitreps.txt:119]: I don't understand your question.

Effects run on the server based on the actual universe, not the universe as seen by a particular player. If you don't add a visibility test to some effect, it won't add it automatically.

Comment (agrrr3) [reply default/customizations/custom_sitreps.txt:119]: That makes more sense to me. So probably I had some other condition which limited the results. If not there should have been a lot of sitreps about system i did not know of yet.

Comment (geoffthemedio) [reply default/customizations/custom_sitreps.txt:119]: Not about systems your empire can't see, but about ships that aren't visible to the player's empire that are arriving in systems you can see. The scope already limits it to systems containing objects owned by the empire receiving the sitrep messages.

**Comment by agrrr3** [comment]
Fixed the visibility checks, sitrep label. Rebased.

**Comment by geoffthemedio** [comment]
Looks ok. Works as intended for you?

**Comment by agrrr3** [comment]
Works fine :)

**Pull request merged**